### ISSUE TITLE
Add (disable-able) warning when user attempts to save machine labels layer to avoid loss of work

### ISSUE THREAD

**Issue by aspence** [DeepLabCut/napari-deeplabcut#172]
I'm pretty sure this is the behavior of napari-deeplabcut on refinement:
1. If you open a folder with machine labels and collecteddata layers
2. The machine labels layer is highlighted by default if loaded at top (mine does this), or the user may have clicked this layer
3. You hit Save or Command-S on mac.
4. It overwrites the machine layer label points into the collecteddata layer points.

From the docs I see that this is the intentional behavior for when you are refining, and it makes sense.

But this can be a problem because it is very easy to completely refine the file, then save and close it and then reload it for inspection later, then out of habit hit command-s for save with the machine labels layer highlighted, which overwrites the collecteddata with any unrefined machine labels points, because machine-labels are never updated, so even if they were moved during the refinement process and are at that point intentionally saved onto the collecteddata, upon reload they are back to their original locations and will overwrite corrected collecteddata.

What about a confirm dialog that pops up when you hit save for the machinelabels layer, and says, "This operation will overwrite collecteddata with all machine layer points; if you have re-loaded an old folder they are likely reverted to non-refined positions and you may lose your work; continue?"

A check box at top right could disable this dialog for experienced users.

Thanks for the great work!

**Comment by C-Achard** [comment]
Hi @aspence, thanks for your comment! 
Yes, that is correct, and in fact several of the safeguards you suggested fit neatly into the plugin refactor that is underway.

Specifically, based on your feedback, I've opened #173, which refines the overwrite warning dialogs that were previously added.

Two main points in the refactor are relevant here:

- As you mentioned, data is never saved to machine layers directly
  - This is intentional and meant to avoid storing human GT/corrections in machine layers 
  - If there is no suitable CollectedData it now prompts the user for a name.
- **When any Points data (machine or otherwise) would overwrite what is already in an h5**, you get a popup saying which file is being overwritten, and which frame(s)/keypoint(s) are affected
  - This can be disabled with a toggle to avoid repeated interruptions while labeling, as suggested

<img width="295" height="511" alt="Image" src="https://github.com/user-attachments/assets/28c4b106-a8da-460d-b563-dc98fbd9f473" />

See [here ](https://github.com/DeepLabCut/napari-deeplabcut/blob/7dcd223767bd67de9b1cc565de4a2600557c60ba/src/napari_deeplabcut/_widgets.py#L1327) and [here](https://github.com/DeepLabCut/napari-deeplabcut/blob/7dcd223767bd67de9b1cc565de4a2600557c60ba/src/napari_deeplabcut/_widgets.py#L1355) for a high-level overview of what happens in the code now when layers are saved, if relevant.

A couple limitations for now:

- The “Skip overwrite confirmations” toggle is currently persisted across sessions as a user setting; I may change that default in the future to reduce accidental overwrites when opening fresh data.
- We still lack a dedicated way to **choose which h5 to save to.** This is something I will work on in the future as I expect it would be useful.

Overall there is still a lot of consolidation, testing and documentation work to be done for the refactor, but I'm happy to address any feedback or feature requests while it is ongoing, so please let me know.

Best,
Cyril

**Comment by aspence** [comment]
This will all be awesome! Thank you! Look forward to trying it out and will try to give helpful feedback. Actually, can I pull the version with these features and play with it? I'm on dlc3.0.0rc13, not sure about which napari-deeplabcut I got with that, but i don't think it has the refactor in yet.

In general, I recently upgraded our aging GPU machines to Ubuntu 24 and dlc3.0.0rc13 with pytorch and it's been really nice. Still took some fiddling with the numerical stack packages and separately the napari stack packages to get both training and napari to work, but once going it's a really nice setup, and so much better than all the tensorflow minutiae.

**Comment by C-Achard** [comment]
Yes, absolutely, simply run (strongly recommend installing in a **separate test environment** to avoid disrupting existing workflows) :
`pip install "git+https://github.com/DeepLabCut/napari-deeplabcut.git@cy/refactor-overwrite-warns"`

This will install the latest version in #173. (Note that the refactor is a WIP, so it's only on GitHub for now.)

Just a bit of a disclaimer:
- **Please backup your data** before trying it out
- Integration with the main DLC GUI is not guaranteed at this stage
- I've added some new features such as dynamically populating the color scheme from the current points layer, which may not scale well yet for very large datasets
- Some bugs may still occur

That said, if you do happen to run into any such issues and can report them, that would help the refactor a lot.

Thanks again !

**Comment by C-Achard** [comment]
(Did a quick test on DLC GUI rc13, seems to work properly)

**Comment by C-Achard** [comment]
Hello, this is now fixed on main thanks to #175. I will close this for now, please re-open if you have further feedback. Thanks!
